### DISCUSSION TITLE
Strange behaviour with Racechrono

### DISCUSSION THREAD

**Discussion by mnmsmushy** [renatobo/bonogps#39] [Q&A]
Hi! Firstly thank you so much for your work towards this and making it possible for a newbie like me to (almost) make my own high refresh rate GPS device.

I tried following the guides to get my device to work with Racechrono. I’m using a M10 with ESP32 board, and everything seems to work except for a really slow gps speed refresh rate
![9B4F2B29-41FD-48F4-AE80-643D47BB6FAA](https://github.com/renatobo/bonogps/assets/16767901/b6c47809-ec68-4aed-8203-8cc5d9d92db2)

These are the messages im seeing from the gps when connected to my laptop thru FTDI.

in the RaceChrono app, latitude and longitude is being updated properly at high refresh rates, just gps speed is totally erratic updating at a rate far below 1hz.

RaceChrono dev says gps speed is read directly from the messages but I don’t know what I’m doing wrong. Could you point me in the right direction?

**Comment by renatobo** [comment]
Hi there!

For RaceChrono, GGA+RMC messages are enabled (as documented in https://github.com/renatobo/bonogps/tree/main/software/connecting/racechrono ) per recommendations available in [Forum message #1](https://racechrono.com/forum/discussion/comment/11252/#Comment_11252) and [Forum message 2](https://racechrono.com/forum/discussion/1421/best-settings-for-qstarz818xt)

RMC includes speed as part of its message ( [format](https://shadyelectronics.com/gps-nmea-sentence-structure/) ) and it seems populated in your screenshot.

Have you tried another App to validate whether it's HW or SW?

**Reply by mnmsmushy** [reply]
Hi! Thank you so much for your software!
Yeah it seems to work well with Harry’s lap timer with the exact same settings.
Based on my screenshot, can I say i set it up correctly?

**Reply by renatobo** [reply]
Are you using WiFi or BT? 
I'd suggest testing at 5 Hz, if that's stable it might be a GPS to ESP32 connectivity issue - I would exclude a GPS issue since it works with HLT.

**Reply by mnmsmushy** [reply]
I tried both wifi and BT actually.. ok I will try RaceChrono at 5hz!
